### ISSUE TITLE
--enable-cusparse=no does not work

### ISSUE THREAD

**Issue by lahwaacz** [hypre-space/hypre#688]
Building hypre v2.25.0 with the `--enable-cusparse=no` configure flag does not work, it fails with the following error:

```
par_ilu.c(2897): error: identifier "hypre_DeviceDataCusparseHandle" is undefined

par_ilu.c(2976): error: identifier "hypre_DeviceDataCusparseHandle" is undefined

par_ilu.c(3069): error: identifier "hypre_DeviceDataCusparseHandle" is undefined

par_ilu.c(3228): error: identifier "hypre_DeviceDataCusparseHandle" is undefined

par_ilu.c(3304): error: identifier "hypre_DeviceDataCusparseHandle" is undefined

5 errors detected in the compilation of "par_ilu.c".
make[1]: *** [../config/Makefile.config:66: par_ilu.obj] Error 1
make[1]: *** Waiting for unfinished jobs....
par_ilu_setup.c(1534): error: identifier "hypre_DeviceDataCusparseHandle" is undefined

par_ilu_setup.c(1535): error: identifier "hypre_CSRMatrixGPUMatDescr" is undefined

par_ilu_setup.c(1538): error: identifier "hypre_SortCSRCusparse" is undefined

par_ilu_setup.c(1636): error: identifier "hypre_DeviceDataCusparseHandle" is undefined

par_ilu_setup.c(2694): error: identifier "hypre_DeviceDataCusparseHandle" is undefined

5 errors detected in the compilation of "par_ilu_setup.c".
make[1]: *** [../config/Makefile.config:66: par_ilu_setup.obj] Error 1
```

There might be more problems as the build did not get to the end...

**Comment by victorapm** [comment]
Hi @lahwaacz,

cusparse is currently required when building hypre with CUDA support. I have a PR that allows building hypre with CUDA and without cusparse, however the `hypre_ILU` solver will not work on the GPU in that scenario. I will let you know once the PR is merged into master

**Comment by lahwaacz** [comment]
@victorapm, thanks for the response. I think the build should definitely succeed with `--enable-cusparse=no` because that is the point of having the option, right? Can you make a feature table documenting which functionality requires which libraries?

**Comment by victorapm** [comment]
@lahwaacz, I updated hypre's build as mentioned above. Let us know if you still have problems. I'll bring up your idea at our developer's meeting, I think it's a good one. Thanks!
